docs/user/azure/install_upi_azurestack.md

```diff
@@ -0,0 +1,591 @@
+
+The deployment steps will read the Red Hat Enterprise Linux CoreOS virtual hard disk (VHD) image and the bootstrap ignition config file
+from a blob. Create a storage account that will be used to store them and export its key as an environment variable.
+
+```sh
+az storage account create -g $RESOURCE_GROUP --location $AZURE_REGION --name ${CLUSTER_NAME}sa --kind Storage --sku Standard_LRS
+export ACCOUNT_KEY=`az storage account keys list -g $RESOURCE_GROUP --account-name ${CLUSTER_NAME}sa --query "[0].value" -o tsv`
+```
+
+### Copy the cluster image
+
+In order to create VMs, the RHCOS VHD must be available in the Azure Stack
+environment. The VHD should be downloaded, decompressed, and uploaded to a
+storage blob.
+
+First, download and decompress the VHD, note that the decompressed file is 16GB:
+
+```sh
+$ export COMPRESSED_VHD_URL=$(curl -s https://raw.githubusercontent.com/openshift/installer/release-4.9/data/data/rhcos-amd64.json | jq -r '(.baseURI + .images.azurestack.path)')
+$ wget $COMPRESSED_VHD_URL
+--2021-08-05 14:40:05--  https://rhcos-redirector.apps.art.xq1c.p1.openshiftapps.com/art/storage/releases/rhcos-4.9/49.84.202107010027-0/x86_64/rhcos-49.84.202107010027-0-azurestack.x86_64.vhd.gz
+Resolving rhcos-redirector.apps.art.xq1c.p1.openshiftapps.com (rhcos-redirector.apps.art.xq1c.p1.openshiftapps.com)... 3.229.121.196, 3.232.64.216, 3.225.72.82, ...
+Connecting to rhcos-redirector.apps.art.xq1c.p1.openshiftapps.com (rhcos-redirector.apps.art.xq1c.p1.openshiftapps.com)|3.229.121.196|:443... connected.
+HTTP request sent, awaiting response... 200 OK
+Length: 1030872938 (983M) [application/gzip]
+Saving to: ‘rhcos-49.84.202107010027-0-azurestack.x86_64.vhd.gz’
+$ gunzip rhcos-49.84.202107010027-0-azurestack.x86_64.vhd.gz
+```
+
+Next, create a container for the VHD:
+
+```sh
+az storage container create --name vhd --account-name ${CLUSTER_NAME}sa
+```
+
+As mentioned above, the VHD size is massive. If you have a fast upload speed,
+you may be satisfied to upload the whole 16GB file:
+
+```sh
+az storage blob upload --account-name ${CLUSTER_NAME}sa --account-key $ACCOUNT_KEY -c vhd -n "rhcos.vhd" -f rhcos-49.84.202107010027-0-azurestack.x86_64.vhd
+```
+
+A better option would be to use [azure-vhd-utils][azure-vhd-utils], which 
+is designed specifically to upload VHD files to Azure:
+
+```sh
+  azure-vhd-utils upload \
+  --localvhdpath rhcos-49.84.202107010027-0-azurestack.x86_64.vhd \
+  --stgaccountname ${CLUSTER_NAME}sa \
+  --stgaccountkey $ACCOUNT_KEY \
+  --containername vhd \
+  --blobname "rhcos.vhd"
+```
+
+### Upload the bootstrap ignition
+
+Create a blob storage container and upload the generated `bootstrap.ign` file:
+
+```sh
+az storage container create --name files --account-name "${INFRA_ID}sa" --public-access blob --account-key "$ACCOUNT_KEY"
+az storage blob upload --account-name "${INFRA_ID}sa" --account-key "$ACCOUNT_KEY" -c "files" -f "bootstrap.ign" -n "bootstrap.ign"
```

Comment:
Should we stick with `--account-name ${CLUSTER_NAME}sa` for the account name, as was done in the other storage-related account setup commands?